app-backend/tile/src/main/scala/tools/ToolTiles.scala

```diff
@@ -0,0 +1,399 @@
+package com.azavea.rf.tile.tools
+
+import java.util.UUID
+
+import com.azavea.rf.common.cache._
+import com.azavea.rf.common.cache.kryo.KryoMemcachedClient
+import com.azavea.rf.common.utils.TileUtils
+import com.azavea.rf.common.KamonTraceRF
+import com.azavea.rf.database.Database
+import com.azavea.rf.database.tables._
+import com.azavea.rf.datamodel._
+import com.azavea.rf.tile.project._
+import com.azavea.rf.tool.ast._
+import com.azavea.rf.tool.eval._
+import com.azavea.rf.tool.params._
+
+import akka.dispatch.MessageDispatcher
+import cats.data._
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.syntax._
+import geotrellis.raster._
+import geotrellis.raster.histogram._
+import geotrellis.raster.render._
+import geotrellis.slick.Projected
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.postgres.PostgresAttributeStore
+import geotrellis.spark.io.s3._
+import geotrellis.vector.Extent
+
+import scala.concurrent._
+import scala.util._
+
+trait ToolTiles extends LazyLogging with KamonTraceRF with ProjectTileUtils with TileIO {
+  implicit val database: Database
+  implicit val blockingDispatcher: MessageDispatcher
```

Comment:
`blockingDispatcher` is the only available `ExecutionContext` for any futures for the functions here. The concern would be that many of these CPU bound and flow control Futures created here are going to be wait-listed by long blocking IO futures. This could prevent requests that already have IO needs satisfied from completing the `flattenTiles` call (for instance) and the request.